distributed/bokeh/tasks/server_lifecycle.py

```diff
@@ -32,32 +35,34 @@
                'http-port': 9786}
 
 
-
 @gen.coroutine
 def task_events(interval, deque, times, index, rectangles, workers, last_seen):
     i = 0
-    with log_errors():
+    try:
         stream = yield eventstream('%(host)s:%(tcp-port)d' % options, 0.100)
         while True:
-            try:
-                msgs = yield read(stream)
-            except StreamClosedError:
-                break
-            else:
-                if not msgs:
-                    continue
-
-                last_seen[0] = time()
-                for msg in msgs:
-                    if 'compute_start' in msg:
-                        deque.append(msg)
-                        times.append(msg['compute_start'])
-                        index.append(i); i += 1
-                        if msg.get('transfer_start') is not None:
-                            index.append(i); i += 1
-                        if msg.get('disk_load_start') is not None:
-                            index.append(i); i += 1
-                        task_stream_append(rectangles, msg, workers)
+            msgs = yield read(stream)
+            if not msgs:
+                continue
+
+            last_seen[0] = time()
+            for msg in msgs:
+                if 'compute_start' in msg:
+                    deque.append(msg)
+                    times.append(msg['compute_start'])
+                    index.append(i)
+                    i += 1
+                    if msg.get('transfer_start') is not None:
+                        index.append(i)
+                        i += 1
+                    if msg.get('disk_load_start') is not None:
+                        index.append(i)
+                        i += 1
+                    task_stream_append(rectangles, msg, workers)
+    except Exception as e:
```

Comment:
In the case of a `StreamClosedError` we probably don't want to exit without logging the error.